sql/analyzer/validate_create_table.go

```diff
@@ -187,6 +190,31 @@ func validateDropColumn(initialSch, sch sql.Schema, dc *plan.DropColumn) (sql.Sc
 	return newSch, nil
 }
 
+// validateDroppingColumnNotUsedInCheckConstraint validates that the specified column name is not referenced in any of
+// the specified table check constraints that reference any other columns.
+func validateDroppingColumnNotUsedInCheckConstraint(columnName string, checks sql.CheckConstraints) error {
+	for _, check := range checks {
+		hasOtherColumn := false
+		var err error
+		expression.TransformUp(check.Expr, func(e sql.Expression) (sql.Expression, error) {
+			if unresolvedColumn, ok := e.(*expression.UnresolvedColumn); ok {
+				if columnName == unresolvedColumn.Name() {
+					err = sql.ErrCheckConstraintInvalidatedByColumnAlter.New(columnName, check.Name)
+				} else {
+					hasOtherColumn = true
+				}
+			}
+			return e, nil
+		})
+
+		// Only prevent dropping column due to checks iff any check that's referencing this column references other columns
+		if err != nil && hasOtherColumn {
+			return err
+		}
```

Comment:
This could be a bit simpler:
```suggestion
		expression.TransformUp(check.Expr, func(e sql.Expression) (sql.Expression, error) {
			if unresolvedColumn, ok := e.(*expression.UnresolvedColumn); ok {
				if columnName != unresolvedColumn.Name() {
					hasOtherColumn = true
				}
			}
			return e, nil
		})

		// Only prevent dropping column due to checks iff any check that's referencing this column references other columns
		if hasOtherColumn {
			return sql.ErrCheckConstraintInvalidatedByColumnAlter.New(columnName, check.Name)
		}
```